pkg/utils/apiresources_test.go

```diff
@@ -228,144 +228,196 @@ func TestResourceConfigGroupParse(t *testing.T) {
 	}
 }
 
-func TestDisabledResourceConfigMixedParse(t *testing.T) {
-	tests := []struct {
-		input    string
-		disabled []schema.GroupVersionKind
-		enabled  []schema.GroupVersionKind
+func TestResourceConfigMixedParse(t *testing.T) {
+	input := "v1/Node,Pod;networking.k8s.io;apps/v1;authorization.k8s.io/v1/SelfSubjectRulesReview"
+
+	// these are the resources that are in the scope of the user specified input
+	resourcesInUserInput := []schema.GroupVersionKind{
+		{
+			Group:   "networking.k8s.io",
+			Version: "v1beta1",
+			Kind:    "Ingress",
+		},
+		{
+			Group:   "networking.k8s.io",
+			Version: "v1",
+			Kind:    "IngressClass",
+		},
+		{
+			Group:   "",
+			Version: "v1",
+			Kind:    "Node",
+		},
+		{
+			Group:   "",
+			Version: "v1",
+			Kind:    "Pod",
+		},
+		{
+			Group:   "authorization.k8s.io",
+			Version: "v1",
+			Kind:    "SelfSubjectRulesReview",
+		},
+		{
+			Group:   "apps",
+			Version: "v1",
+			Kind:    "HealthState",
+		},
+	}
+
+	// these are the resources that are not in the scope of the user specified input
+	resourcesNotInUserInput := []schema.GroupVersionKind{
+		{
+			Group:   "",
+			Version: "v1",
+			Kind:    "Ingress",
+		},
+		{
+			Group:   "apps",
+			Version: "v1beta1",
+			Kind:    "IngressClass",
+		},
+		{
+			Group:   "authorization.k8s.io",
+			Version: "v1",
+			Kind:    "LocalSubjectAccessReview",
+		},
+	}
+
+	tests := map[string]struct {
+		isAllowList bool
+		disabled    []schema.GroupVersionKind
+		enabled     []schema.GroupVersionKind
 	}{
+		"disabled list": {
+			isAllowList: false,
+			disabled:    resourcesInUserInput,
+			enabled:     resourcesNotInUserInput,
+		},
+		"enabled list": {
+			isAllowList: true,
+			disabled:    resourcesNotInUserInput,
+			enabled:     resourcesInUserInput,
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			r := NewResourceConfig(test.isAllowList)
+			if err := r.Parse(input); err != nil {
+				t.Fatalf("Parse() returned error: %v", err)
+			}
+
+			for i, o := range test.disabled {
+				if ok := r.IsResourceDisabled(o); !ok {
+					t.Errorf("%d: expected resource to be disabled : %v", i, o)
```

Comment:
```suggestion
					t.Errorf("IsResourceDisabled(%v) = false, want true", o)
```
Could you please fix the test code to follow the test output format?

we're following golang best practices here
`Test outputs should output the actual value that the function returned before printing the value that was expected. A usual format for printing test outputs is “YourFunc(%v) = %v, want %v”.`
https://go.dev/wiki/TestComments#got-before-want

`Do not use the index of the test in the test table as a substitute for naming your tests or printing the inputs. Nobody wants to go through your test table and count the entries in order to figure out which test case is failing.`
https://go.dev/wiki/TestComments#identify-the-input

Reply:
I ended up refactoring all the tests in apiresources_test.go, I have added small utility functions to make tests more readable and not repeating code multiple times. please let me know if that looks good.